pkg/server/plugin/datastore/datastore.go

```diff
@@ -40,9 +38,9 @@ type DataStore interface {
 	UpdateAttestedNode(context.Context, *common.AttestedNode, *common.AttestedNodeMask) (*common.AttestedNode, error)
 
 	// Node selectors
-	GetNodeSelectors(context.Context, *GetNodeSelectorsRequest) (*GetNodeSelectorsResponse, error)
+	GetNodeSelectors(ctx context.Context, spiffeID string, tolerateStale bool) (*NodeSelectors, error)
 	ListNodeSelectors(context.Context, *ListNodeSelectorsRequest) (*ListNodeSelectorsResponse, error)
-	SetNodeSelectors(context.Context, *SetNodeSelectorsRequest) (*SetNodeSelectorsResponse, error)
+	SetNodeSelectors(context.Context, *NodeSelectors) error
```

Comment:
I know it's going to churn up the code a bit, particularly in the tests, but I wonder how you'd feel about getting rid of the NodeSelectors struct.
`GetNodeSelectors` could just return the slice of selectors instead of repeating the SPIFFE ID it was given, since the caller already knows it.
`SetNodeSelectors` could just take the spiffe ID and slice of selectors as parameters.

Reply:
LOL! I've wondered more than once if it would be cleaner to pull that struct apart. I'll look at it...

Reply:
pushed a commit with NodeSelectors struct removed